mustache.js

```diff
@@ -592,12 +610,30 @@
       return this.renderTokens(token[4], context, partials, originalTemplate);
   };
 
+  Writer.prototype.indentPartial = function indentPartial (partial, indentation) {
+    var filteredIndentation = indentation.replace(/[^ \t]/g, '');
+    var partialByNl = partial.split('\n');
+    for (var i = 0; i < partialByNl.length; i++) {
+      if (partialByNl[i].length) {
+        partialByNl[i] = filteredIndentation + partialByNl[i];
+      }
+    }
+    return partialByNl.join('\n');
+  };
+
   Writer.prototype.renderPartial = function renderPartial (token, context, partials, tags) {
     if (!partials) return;
 
     var value = isFunction(partials) ? partials(token[1]) : partials[token[1]];
-    if (value != null)
-      return this.renderTokens(this.parse(value, tags), context, partials, value);
+    if (value != null) {
+      var tagIndex = token[5];
+      var indentation = token[4];
+      var indentedValue = value;
+      if (tagIndex == 0 && indentation) {
+        var indentedValue = this.indentPartial(value, indentation);
```

Comment:
Only nitpick that got my brain worried at first; this kind of re-declaration of an already existing variable will explode when using `let | const`. Although that doesn't apply to `var` which is used here, I'm a fan of avoiding those kinds of questions from the get-go..

Bottom line removing `var` from this line would have avoided my initial worries and even make the intended purpose more clear (at least in my eyes).

Or did you very deliberately use `var` here for reasons I'm not aware of?

```suggestion
        indentedValue = this.indentPartial(value, indentation);
```

Reply:
I agree, this was part of kevindew's commit which I didn't noticed. 
I am committing the suggestion